status-counter.js

```diff
@@ -12,66 +15,53 @@ class CounterWrap extends Wrap {
-   * @param {String} params.version Version for counter table
+   * @param {Object} key Object containing key of item to increment
    * @param {Number} amount Number to increment
    * @returns {Promise} when operation complete
    * @public
    */
-  async increment(params, amount = 1) {
-    const client = await this._getConnection();
-    const query = `UPDATE status_counter SET count=count+${amount} WHERE pkg=:pkg AND env=:env AND version=:version`;
-    return client.execute(query, params, { prepare: true, counter: true });
+  async increment(key, amount = 1) {
+    return this.update({ ...key, count: { $add: amount } });
   }
   /**
    * Decrement the given counter for this table in the manual way with the
    * cassandra driver until we put this into datastar in a proper way
    *
    * @function increment
-   * @param {Object} params Parameters for performing increment
-   * @param {String} params.env Env for Counter table
-   * @param {String} params.pkg Pkg for Counter table
-   * @param {String} params.version Version for counter table
+   * @param {Object} key Object containing key of item to increment
    * @param {Number} amount Number to increment
    * @returns {Promise} when operation complete
    * @public
    */
-  async decrement(params, amount = 1) {
-    const client = await this._getConnection();
-    const query = `UPDATE status_counter SET count=count-${amount} WHERE pkg=:pkg AND env=:env AND version=:version`;
-    return client.execute(query, params, { prepare: true, counter: true });
+  async decrement(key, amount = 1) {
+    return this.update({ ...key, count: { $add: -1 * amount } });
   }
 }
 
 /**
- * Returns the wrapped Datastar model for status_counter
+ * Returns a Wrapped Dynastar model for status_counter
  * This model is used for counting the completions of each locale for a given
  * build so we can track overall progress
  *
  * @function statcount
- * @param {Datastar} datastar Datastar instance
+ * @param {Object} dynamo Dynamo object model
  * @returns {CounterWrap} model for StatusCounter
  * @private
  */
-module.exports = function statcount(datastar) {
-  const cql = datastar.schema.cql;
-  const StatusCounter = datastar.define('status_counter', {
-    schema: datastar.schema.object({
-      pkg: cql.text(),
-      env: cql.text(),
-      version: cql.text(),
-      count: cql.counter()
-    }).partitionKey(['pkg', 'env', 'version']),
-    with: {
-      compaction: {
-        class: 'LeveledCompactionStrategy'
-      }
+module.exports = function statcount(dynamo) {
+  const hashKey = 'key';
+  const model = dynamo.define('status_counter', {
+    hashKey,
+    tableName: 'status_counter',
+    schema: {
+      key: Joi.string(),
+      pkg: Joi.string(),
+      env: Joi.string(),
+      version: Joi.string(),
+      count: Joi.number()
     }
   });
-
-  return new CounterWrap(StatusCounter);
+  return new CounterWrap(new Dynastar({ model, hashKey }));
```

Comment:
in this case hashKey doesn't HAVE to be defined since its the default in `dynastar` but its can be here.

```suggestion
  return new CounterWrap(new Dynastar({ model, hashKey, createKey }));
```